### PR TITLE
Introduce observation models and adapter

### PR DESCRIPTION
**Pull request by fridex**
Depends-On: https://github.com/goern/janusgraph-openshift/pull/24

### CODE REVIEW

**Comment by fridex** [comment]
recheck

File: thoth/storages/graph/models.py [hunk 1]
```diff
@@ -85,9 +85,31 @@ class RuntimeEnvironment(VertexBase):
...
 
 class SoftwareStack(VertexBase):
-    """Observations we have about the given stack."""
+    """A software stack crated by packages in specific versions."""
 
-    # TODO: add observation info and info about origin
+    installable = VertexProperty(properties.Boolean, default=True)
+
+
+class SoftwareStackObservation(VertexBase):
```
Comment (goern) [inline thoth/storages/graph/models.py]: do we have a PackageVersionObservation?!

File: thoth/storages/graph/models.py [hunk 2]
```diff
@@ -85,9 +85,31 @@ class RuntimeEnvironment(VertexBase):
...
+    performance_index = VertexProperty(properties.Float)
+    observations_document_id = VertexProperty(properties.String)
+
+
+class HardwareInformation(VertexBase):
+    """Hardware specification and propertires."""
+
+    cpu_vendor = VertexProperty(properties.String)
+    cpu_model_name = VertexProperty(properties.String)
+    cpu_cores = VertexProperty(properties.Integer)
```
Comment (goern) [inline thoth/storages/graph/models.py]: we need cpu_model:unsighed_int and cpu_family:unsigned_int too :)

@sub-mod can you give some advice what is required for TF build/run?

**Comment by fridex** [comment]
On Tue, Sep 25, 2018, 13:07 Christoph Görn <notifications@github.com> wrote:

> *@goern* commented on this pull request.
> ------------------------------
>
> In thoth/storages/graph/models.py
> <https://github.com/thoth-station/storages/pull/119#discussion_r220148178>
> :
>
> >
> -    # TODO: add observation info and info about origin
> +    installable = VertexProperty(properties.Boolean, default=True)
> +
> +
> +class SoftwareStackObservation(VertexBase):
>
> do we have a PackageVersionObservation?!
>

No. The reason why I created a separate node: software stack observations
are dependent on HW, runtime environment and the actual software stack (see
updated schema). Any other ideas?

> ------------------------------
>
> In thoth/storages/graph/models.py
> <https://github.com/thoth-station/storages/pull/119#discussion_r220148684>
> :
>
> > +    installable = VertexProperty(properties.Boolean, default=True)
> +
> +
> +class SoftwareStackObservation(VertexBase):
> +    """Observations we have about the given stack based on run on a specific hardware."""
> +
> +    performance_index = VertexProperty(properties.Float)
> +    observations_document_id = VertexProperty(properties.String)
> +
> +
> +class HardwareInformation(VertexBase):
> +    """Hardware specification and propertires."""
> +
> +    cpu_vendor = VertexProperty(properties.String)
> +    cpu_model_name = VertexProperty(properties.String)
> +    cpu_cores = VertexProperty(properties.Integer)
>
> we need cpu_model:unsighed_int and cpu_family:unsigned_int too :)
>

Why integers?

**Comment by goern** [comment]
I think its uint because of https://github.com/gcc-mirror/gcc/blob/master/gcc/config/i386/cpuid.h#L256

**Comment by fridex** [comment]
@sub-mod could you please have a look? do we need something more here?
@goern adjusted based on review comments

**Comment by goern** [comment]
recheck

**Pull request merged**